src/Silverback.Integration.Kafka/Messaging/Configuration/KafkaBrokerHeadersEnricher.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using Silverback.Diagnostics;
+using Silverback.Messaging.Messages;
+
+namespace Silverback.Messaging.Configuration
+{
+    internal sealed class KafkaBrokerHeadersEnricher
+        : IBrokerHeadersEnricher<KafkaProducerEndpoint>, IBrokerHeadersEnricher<KafkaConsumerEndpoint>
+    {
+        public IEnumerable<MessageHeader> GetCustomErrorHeaders(
+            IRawInboundEnvelope envelope,
+            Exception exception)
+        {
+            var customHeaders = new List<MessageHeader>
+            {
+                new(KafkaMessageHeaders.SourceTopic, envelope.ActualEndpointName),
+                new(
+                    KafkaMessageHeaders.SourceConsumerGroupId,
+                    ((KafkaConsumerEndpoint)envelope.Endpoint).Configuration.GroupId),
+                new(DefaultMessageHeaders.FailureReason,
+                    $"{exception.GetType().FullName} in {exception.Source}"),
+                new(KafkaMessageHeaders.SourcePartition, envelope.BrokerMessageIdentifier.Key),
```

Comment:
Don't use `Key` and `Value` but cast to the correct type `KafkaOffset` and use the proper values (e.g. `KafkaOffset.Partition`).

Reply:
Done, thanks!